src/vs/base/parts/storage/node/storage.ts

```diff
@@ -20,6 +20,7 @@ interface IDatabaseConnection {
 }
 
 export interface ISQLiteStorageDatabaseOptions {
+	readonly intent?: number;
```

Comment:
I would prefer if we simply had the option to open the DB as `readonly` and not expose the underlying SQLite modes as option. I think in the end we just have 1 client that wants to open the DB readonly so I would optimise for that scenario.